pkg/kubectl/genericclioptions/printers/name.go

```diff
@@ -50,13 +50,6 @@ func (p *NamePrinter) PrintObj(obj runtime.Object, w io.Writer) error {
 	}
 
 	if meta.IsListType(obj) {
-		// we allow unstructured lists for now because they always contain the GVK information.  We should chase down
-		// callers and stop them from passing unflattened lists
-		// TODO chase the caller that is setting this and remove it.
-		if _, ok := obj.(*unstructured.UnstructuredList); !ok {
```

Comment:
This change was introduced in https://github.com/kubernetes/kubernetes/commit/420dd9bf667585cc516e25909c21a4c7ed7797c7.

I'm not sure we disallow `v1.List` here. We flatten and handle objects individually:

https://github.com/kubernetes/kubernetes/blob/330cf37e0ae2e81a3398972112f154891ab68708/pkg/kubectl/cmd/convert.go#L223-L226

Then combine them into a list if a single object is asked:

https://github.com/kubernetes/kubernetes/blob/330cf37e0ae2e81a3398972112f154891ab68708/pkg/kubectl/cmd/convert.go#L184-L188

This should ensure that the objects contain the correct gvk, right?

Reply:
I think `v1.List` could also be accepted here. @deads2k, thoughts?

Typed lists (e.g. `v1.PodList`) do not always contain GVK info on individual items (inspect the response of the apiserver of fetching `/api/v1/pods`)

Reply:
> Typed lists (e.g. v1.PodList) do not always contain GVK info on individual items (inspect the response of the apiserver of fetching /api/v1/pods)

Yes, then it makes sense to just allow `v1.List`. :+1: I'll wait for @deads2k to comment before making the change.

But out of curiosity - why do the individual items not contain GVK for typed lists?

Reply:
> I think v1.List could also be accepted here, but I'm not sure I'd open it up to all list types. @deads2k, thoughts?

UnstructuredLists constain Unstructureds which will always have a recongizable GVK.  v1.List contains runtime.RawExtension which is only guaranteed to have a .Raw which does not guarantee a recongizeable GVK.  The Object is optional.

I'm strongly disinclined to change a printer for the sake of this one command and to do so in a way that does not guarantee that you can print the values requested.  Instead I would ask why the Convert command has options related to list-ifying things.

Reply:
And also why the convert command even has an option print names.  When is that useful?

Reply:
I would be in favor of removing the Name printer from this command, and therefore removing support for `-o name` output there.

An alternative we could also consider for the time being, could be to iterate through all objects in the list (only when name printing), and pass those individually to the printer (similarly to how it is done in: https://github.com/openshift/origin/blob/master/pkg/oc/cli/process/process.go#L438)